### PR TITLE
updating formatNumber to handle currencies

### PR DESCRIPTION
**Pull request by matthewwhitaker**
Updating `formatNumber` method to handle currencies based on the set locale. 

`formatNumber` now uses `Intl` instead of `numeral` and no longer takes  `format` as an argument.

Updated changelog, package version and Travis node version.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -28,10 +28,13 @@ export default class I18n {
...
-   * Leverages `numeral` for number formatting
+   * Leverages Intl.NumberFormat for currency formatting
    */
-  formatNumber (format: string, number: number): string {
-    return numeral(number).format(format)
+  formatNumber (number: number, style?: string, currency?: string): string {
+    const params = {}
+    if (style) params.style = style
+    if (currency) params.currency = currency
+    return new global.Intl.NumberFormat(this.locale, { style: style, currency: currency }).format(number)
```
Comment (andresgutgon) [inline src/index.js]: Support for this part of Intl spec looks pretty good in all browsers. Anyway just wondering if we should ship this package with an Intl pollyfill. 
Opinions @fcsonline @masylum ?

File: package.json [hunk 2]
```diff
@@ -1,6 +1,6 @@
 {
   "name": "factorial-i18n",
-  "version": "0.1.0",
+  "version": "0.2.0",
```
Comment (andresgutgon) [inline package.json]: This is definitely a mayor version change :trollface:

Comment (andresgutgon) [reply package.json]: Also are we using now `numeral` for something? if not remove from dependencies

File: .travis.yml [hunk 3]
```diff
@@ -1,3 +1,3 @@
 language: node_js
 node_js:
-  - "6.3.0"
+  - "13.1.0"
```
Comment (fcsonline) [inline .travis.yml]: LOL. 7 years of development in one line

File: package.json (hunk 2 above)
Comment (matthewwhitaker) [reply package.json]: It's still being used in `setLocale`, but I haven't investigated properly whether it still needs to be used. I can take a look if you think there's a big advantage in removing it from dependencies.

File: src/index.js (hunk 1 above)
Comment (masylum) [reply src/index.js]: Intl polyfill is a drama. This is supported for evergreen and IE11+ browsers so I wouldn't complicate it.

File: package.json (hunk 2 above)
Comment (matthewwhitaker) [reply package.json]: Removed numeral 👍

File: src/index.js [hunk 4]
```diff
@@ -20,18 +18,16 @@ export default class I18n {
...
-    }
   }
 
   /**
-   * Leverages `numeral` for number formatting
+   * Leverages Intl.NumberFormat for currency formatting
    */
-  formatNumber (format: string, number: number): string {
-    return numeral(number).format(format)
+  formatNumber (number: number, style?: string, currency?: string): string {
```
Comment (andresgutgon) [inline src/index.js]:
```javascript
type BaseFormat = {
  number: number
}
type FormatCurrency = BaseFormat & {
  number: number
  style: 'currency',
  currency: 'EUR' | 'DOLLAR'
}
type OtherStyles = BaseFormat & { 
  style: 'decimal' | 'percent' | 'unit' 
}
type FormatOptions = BaseFormat | OtherStyles | CurrencyStyle
formatNumber (number: number, { style, currency }) {
}
```

[Repel example](https://flow.org/try/#0PQKgBAAgZgNg9gdzCYAoALgTwA4FMwBicATgLYCG6AwgK7HG4B2AxpmALxgDeqYYAzlhi4AXGADkzOgxaZxAGl5gp9JqzHiAogFUASuLAAfCQBEA8gBkLAQX2oAvhhz4z6ABa5iAZSG5+HbjAlQUxhDQATXGYASwoYA2NxPGJmJnQEiRpGaPSgxyw8QhIKdAA5GlIAI08zbHRouEZ-TiIySlpVWSMwVw9vX35UKCzmesawKGLKcqrPMAAKRgrq4jEl2eJ5MDg6hqaAfjFWkpmV2rGmgEpuJWZGwQmp9HO95u3d+6NjLkc+RlwkABzeCVcgwAB0AElGOgIadPMdKPNxH4FO8LvxLuDJm10Itlp5Lg5UEMnvDiPMAMyUrZcAS+CJRWJg8T2Ik4k4EinU2n00KiCTJVIw1nsslcqk0wIhMKZbLpNmk3HkyW8mUCyTSNRyLYqGTqCQ6fRgNlAA)

File: __tests__/index.spec.js [hunk 5]
```diff
@@ -92,9 +92,27 @@ describe('i18n', () => {
...
+    })
+    it('formats the correct currency without locale code', () => {
+      expect(i18n.formatNumber(1500, 'currency', 'GBP')).toBe('£1,500.00')
+    })
+    it('formats number with just a number', () => {
+      expect(i18n.formatNumber(1500)).toBe('1,500')
+    })
+    it('formats currency with setLocale of language and country', () => {
+      i18n.setLocale('de-DE')
+      const spacer = String.fromCharCode(160)
```
Comment (gtrias) [inline __tests__/index.spec.js]: what a drama :sweat_smile:

File: src/index.js (hunk 4 above)
Comment (masylum) [reply src/index.js]: @andresgutgon you are making a lot of assumptions here. Typescript itself marks `style` and `currency` as `string` for instance:

https://github.com/Microsoft/TypeScript/blob/master/src/lib/es5.d.ts#L4216

Comment (masylum) [reply src/index.js]: I wouldn't complicate it for now

Comment (andresgutgon) [reply src/index.js]: Ok but we're not going to support all currencies in the world. Only a subset. Also this project use Flow isn't it?

With that typing we diferienciate between 3 ways of using this method.
1. No options
2. Format a 'decimal' | 'percent' | 'unit' 
3. Format a currency (the once we support)

But as I said totally fine to ignore this if you think doesn't make sense

Comment (andresgutgon) [reply src/index.js]: LOOOOL
image


Eso no vale. Switch to TS in the same PR 😂

**Pull request merged**